niworkflows/viz/utils.py

```diff
@@ -75,19 +74,24 @@ def svg_compress(image, compress='auto'):
 
     # Compress the SVG file using SVGO
     if (_which('svgo') and compress == 'auto') or compress == True:
+        cmd = 'svgo -i - -o - -q -p 3 --pretty --disable=cleanupNumericValues'
         try:
-            p = subprocess.run(
-                "svgo -i - -o - -q -p 3 --pretty --disable=cleanupNumericValues",
-                input=image.encode('utf-8'), stdout=subprocess.PIPE,
-                shell=True, check=True)
-        except FileNotFoundError:
-            if compress is True:
-                raise
+            if PY3:
+                pout = subprocess.run(cmd, input=image.encode('utf-8'), stdout=subprocess.PIPE,
+                                      shell=True, check=True).stdout
+            else:
+                p = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE,
+                                     stdout=subprocess.PIPE)
+                pout, _ = p.communicate(input=image)
```

Comment:
Since we can't use `check=True`, we'll need to check the return code, ourselves:

```Python
if p.returncode != 0:
    raise subprocess.CalledProcessError(p.returncode, cmd)
```

And I think you want `input=image.encode('utf-8')`? Or is that not necessary in Py2?

Sorry to make the suggestion (feel free to say no), but would it be easier to do something at the top that mimics what we need from the `run` API? Like (I haven't run or tested this):

```Python
try:
    from subprocess import run
except ImportError:
    def run(args, input=None, stdout=None, stderr=None, shell=False, check=False):
        from collections import namedtuple
        result = namedtuple('result', ('stdout', 'stderr'))
        proc = subprocess.run(args, stdout=stdout, shell=shell, check=check)
        res = result(proc.communicate(input=input))
        if check and proc.returncode != 0:
            raise subprocess.CalledProcessError(proc.returncode, args)
        return res
```

Then the main changes for the rest of the code are `FileNotFoundError -> OSError`.